src/js/actions/export.js

```diff
@@ -0,0 +1,365 @@
+            .bind(this)
+            .then(function (pathArray) {
+                // Do we need to be aware of exports that return >1 file path?
+                var assetProps = {
+                    filePath: pathArray[0],
+                    status: ExportAsset.STATUS.STABLE
+                };
+                return this.transfer(updateLayerExportAsset, document, layer, assetIndex, assetProps);
+            });
+    };
+
+    /**
+     * Export all assets for the given document for which export has been enabled (layer.exportEnabled)
+     *
+     * @param {Document} document
+     * @return {Promise} resolves when all assets have been exported
+     */
+    var exportAllAssets = function (document) {
+        if (!document) {
+            Promise.resolve("No Document");
+        }
+
+        if (!_exportService || !_exportService.ready()) {
+            Promise.reject("Export Service is not available");
+        }
+
+        var documentID = document.id,
+            documentExports = this.flux.stores.export.getDocumentExports(documentID),
+            layerExportsMap = documentExports && documentExports.layerExportsMap,
+            exportArray = [];
+
+        if (!layerExportsMap || layerExportsMap.size < 1) {
+            return Promise.resolve("no assets to export");
+        }
+
+        // Iterate over the exports map, find the associated layer, test of "exportEnabled"
+        layerExportsMap.forEach(function (layerExportAssets, layerID) {
+            var layer = document.layers.byID(layerID);
+            if (layer.exportEnabled) {
+                // Export all assets for this layer
+                layerExportAssets.forEach(function (asset, index) {
+                    exportArray.push(_exportAsset.call(this, document, layer, index, asset));
+                }, this);
+            }
+        }, this);
+
+        return Promise.all(exportArray);
+    };
+    exportAllAssets.reads = [locks.JS_DOC];
+    exportAllAssets.writes = [locks.PS_GENERATOR];
+    exportAllAssets.transfers = [updateLayerExportAsset];
+
+    /**
+     * Handle the standard onReset action
+     *
+     * @return {Promise}
+     */
+    var onReset = function () {
+        /*
+        // Not sure if we need to do ANYTHING upon reset.  Closing the connection was a pretty bad idea, natch
+        _exportService.close();
+        _exportService = null;
+        */
+        return Promise.resolve();
+    };
+    onReset.reads = [];
+    onReset.writes = [];
+
+    /**
+     * Fetch relevant data from both the document and export stores, and sync the metadata to
+     * the photoshop layer "extension data"
+     *
+     * TODO maybe this should handle a set of layers, for efficiency when called by layer actions?
+     *
+     * @private
+     * @param {number} documentID
+     * @param {number} layerID
+     * @return {Promise}
+     */
+    var _syncLayerExportMetadata = function (documentID, layerID) {
```

Comment:
Can we roughly order the functions in this file in dependency order? Privates up top, publics down below, lifecycle all the way at the bottom?